pypeit/calibrations.py

```diff
@@ -564,61 +590,59 @@ def get_flats(self):
-
-        # Return cached images
-        if self._cached('flatimages', self.master_key_dict['flat']):
-            self.flatimages = self.calib_dict[self.master_key_dict['flat']]['flatimages']
-            self.flatimages.is_synced(self.slits)
-            self.slits.mask_flats(self.flatimages)
-            return self.flatimages
+        illum_image_files, self.master_key_dict['flat'] = self._prep_calibrations('illumflat')
+        pixflat_image_files, self.master_key_dict['flat'] = self._prep_calibrations('pixelflat')
 
         masterframe_filename = masterframe.construct_file_name(flatfield.FlatImages,
                                                            self.master_key_dict['flat'], master_dir=self.master_dir)
         # The following if-elif-else does:
-        #   1.  Try to load a MasterFrame (if reuse_masters is True)
+        #   1.  Try to load a MasterFrame (if reuse_masters is True).  If successful, pass it back
         #   2.  Build from scratch
-        #   3.  Replace the built pixel-flat with user supplied (e.g. LRISb)
+        #   3.  Load any user-supplied images to over-ride any built
 
+        # Load MasterFrame?
         if os.path.isfile(masterframe_filename) and self.reuse_masters:
-            # Load MasterFrame
             self.flatimages = flatfield.FlatImages.from_file(masterframe_filename)
             self.flatimages.is_synced(self.slits)
             self.slits.mask_flats(self.flatimages)
-        elif len(trace_image_files) > 0:
-            # Process/combine the input pixelflat frames
-            # TODO -- Include an illum frametype eventually
-            # TODO This is incorrect nad is only a hack for LRIS-Blue where we want to only run this code
-            # for the illumination flat construction. There needs to be a pixelflat and an illumflat, and
-            # in cases where the files are different, we need to run the flat field code twice.
-            stacked_traceflat = buildimage.buildimage_fromlist(self.spectrograph, self.det,
-                                                    self.par['traceframe'],
-                                                    trace_image_files,
-                                                    bias=self.msbias, bpm=self.msbpm)
+            return self.flatimages
+
+
+        # Generate the image?
+        #if len(pixflat_image_files) > 0:
+        #    stacked_pixelflat = buildimage.buildimage_fromlist(self.spectrograph, self.det,
+        #                                                       self.par['pixelflatframe'],
+        #                                                       pixflat_image_files,
+        #                                                       bias=self.msbias, bpm=self.msbpm)
+        # TODO -- Allow for separate pixelflat and illumflat images
+        if len(illum_image_files) > 0:
+            # CHECK
+            if len(pixflat_image_files) > 0 and illum_image_files != pixflat_image_files:
```

Comment:
Please add a TODO here and a github issue that this needs to be fixed. For example, with LRIS we may want to use sky flats for illumination and dome flats for pixel. Also in that todo we need to always reqiure that the traceimage is the same as the illum image if they are both set. That should probably be done in fitstable.

Reply:
You are better suited to do both.

Reply:
TODO added.